TalkingLeaves.glyphsPlugin/Contents/Resources/TalkingLeaves.py

```diff
@@ -275,11 +276,15 @@ def getLangsForScript_(self, script):
 
         # We just need unsupportedChars but it takes a few steps to get there
         # in somewhat-readable code
-        orthoBase = list(set(ortho['base']))
-        orthoMarks = [m[-1] for m in ortho.get('marks', '').split()]
-        orthoNumerals = list(set(ortho.get('numerals', '')))
-        orthoPunctuation = list(set(ortho.get('punctuation', '')))
-        orthoChars = orthoBase + orthoMarks + orthoNumerals + orthoPunctuation
+        orthography = hyperglot.orthography.Orthography(ortho)
+        orthoBase = sorted(set(orthography.base_chars))
+        orthoMarks = orthography.base_marks
```

Comment:
You can use the `Orthography` convenience object. It's a dict wrapper around the yaml dict, but has a few niceties.

`orthography.base_chars` returns only unicodes, so will have decomposed any unencode base + mark combinations. (I've actually just fixed that this should return only unique unicodes, but alas for now the `set()` wrapper. Returning those `sorted()` yields less busy "missing chars" lists.

`orthography.base_marks` returns only marks that are a) used in base or b) explicitly in `marks` but not in `auxiliary` — the previous code would include marks also if they were only required in `auxiliary`.